src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleRequestSender.cs

```diff
@@ -268,16 +369,48 @@ private bool HandShakeWithVsTestConsole()
                 else if (message.MessageType == MessageType.ProtocolError)
                 {
                     // TODO : Payload for ProtocolError needs to finalized.
-                    EqtTrace.Error("VsTestConsoleRequestSender: Version Check failed. ProtolError was revceived from the runner");
+                    EqtTrace.Error("VsTestConsoleRequestSender.HandShakeWithVsTestConsole: Version Check failed. ProtolError was revceived from the runner");
                 }
                 else
                 {
-                    EqtTrace.Error("VsTestConsoleRequestSender: VersionCheck Message Expected but different message received: Received MessageType: {0}", message.MessageType);
+                    EqtTrace.Error("VsTestConsoleRequestSender.HandShakeWithVsTestConsole: VersionCheck Message Expected but different message received: Received MessageType: {0}", message.MessageType);
                 }
             }
             else
             {
-                EqtTrace.Error("VsTestConsoleRequestSender: SessionConnected Message Expected but different message received: Received MessageType: {0}", message.MessageType);
+                EqtTrace.Error("VsTestConsoleRequestSender.HandShakeWithVsTestConsole: SessionConnected Message Expected but different message received: Received MessageType: {0}", message.MessageType);
+            }
+
+            return success;
+        }
+
+        private async Task<bool> HandShakeWithVsTestConsoleAsync()
+        {
+            var success = false;
+            var message = await this.communicationManager.ReceiveMessageAsync(this.processExitCancellationTokenSource.Token);
+            if (message.MessageType == MessageType.SessionConnected)
+            {
+                this.communicationManager.SendMessage(MessageType.VersionCheck, this.protocolVersion);
+                message = await this.communicationManager.ReceiveMessageAsync(this.processExitCancellationTokenSource.Token);
+
+                if (message.MessageType == MessageType.VersionCheck)
```

Comment:
The switch logic remains same across the sync and async variants of `HandShake`, `Discovery` and `Execution`. Could we extract it to a method to reduce duplication?